src/libsyntax_ext/proc_macro_registrar.rs

```diff
@@ -305,6 +386,23 @@ fn mk_registrar(cx: &mut ExtCtxt,
         cx.stmt_expr(expr)
     }).collect::<Vec<_>>();
 
+    stmts.extend(custom_attrs.iter().map(|ca| {
+        (
+            cx.expr_str(ca.span, ca.function_name.name),
+            cx.path_global(ca.span, vec![ca.function_name]),
+            ca.span
+        )
+    }).map(|(name, path, span)| {
+        let registrar = cx.expr_ident(span, registrar);
+        let ufcs_path = cx.path(span, vec![proc_macro, __internal, registry,
+                                           register_custom_attribute]);
+        cx.expr_call(span,
+                     cx.expr_path(ufcs_path),
+                     vec![registrar, name, cx.expr_path(path)])
+    }).map(|expr| {
+        cx.stmt_expr(expr)
+    }));
```

Comment:
I believe you can use a single map and closure here (compose the current closures).